### PR TITLE
Force Chief::Command result to be a Chief::Result object

### PR DESCRIPTION
**Pull request by smeriwether**
After some discussion we think it is best practice to return a
`Chief::Result` object from all `Chief::Command`s. We think this will lead
to easier to reason about code.

An example:

```ruby
class Example < Chief::Command
  def call
    some_other_chief_command
  end

  private

  def some_other_chief_command
    Command::DoSomething.call
  end
end
```

Code like this leads to a few questions:
- Do we not care about `some_other_chief_command` failing?
- Did we forget to to handle the `some_other_chief_command` fail case?
- Is the caller handling the `some_other_chief_command` error?
- Is the caller doing something with `some_other_chief_commands` value?

By enforcing every `Chief::Command` to call `success!` or `fail!` we can
force the developer to think about all of those questions and we can
help the future developer better understand the command.

### CODE REVIEW

**Review by kellywu — APPROVED** [review]
@smeriwether LGTM!

File: spec/chief/integration_spec.rb [hunk 1]
```diff
@@ -25,7 +25,31 @@ def call
...
+  class ExampleWithChiefResultPassThroughCommand < Command
+    def call
+      ExampleFailingCommand.call(:some_value)
+    end
+  end
+
   RSpec.describe Command do
+    describe "Commands that don't return a Chief::Result object" do
+      it 'should raise an error' do
+        expect { ExampleWithoutChiefResultCommand.call }.to raise_error(RuntimeError)
```
Comment (bradwheel) [inline spec/chief/integration_spec.rb:43]: @smeriwether What do you think about returning a more specific error class here? RuntimeErrors can be misleading when trying to debug.

Comment (stephenmhoc) [reply spec/chief/integration_spec.rb:43]: I thought about that. My reasoning was this:

1. The RuntimeError's message is descriptive
2. Nobody should ever rescue from this error

I don't have strong feelings about this if you do I will certainly change it.

Comment (stephenmhoc) [reply spec/chief/integration_spec.rb:43]: On second thought, I have the same thoughts as described ^ but I think the error message should be changed. I will think of a better message.

Current Message:
`RuntimeError: #call must return a Chief::Result object` 

There are 2 ways for this to fail and the message only talks about one of them
* result isn't a `Chief::Result` object
* command didn't call `success!` or `fail!`

Comment (stephenmhoc) [reply spec/chief/integration_spec.rb:43]: The new error message is:
`{class_name} must call success! or fail! and return the result`

which I am happy with

**Pull request merged**